src/d3d11/d3d11_device.cpp

```diff
@@ -2445,12 +2445,334 @@ namespace dxvk {
+      return nullptr;
+    return (ID3D11SamplerState *)got->second;
+  }
+  void D3D11DeviceExt::AddSrvAndHandleNVX(ID3D11ShaderResourceView *srv_imageview, uint32_t handle) {
+    auto mapslock = m_multithreadMapsLock.AcquireLock();
+    m_SrvHandleToPtr[handle] = srv_imageview;
+  }
+  ID3D11ShaderResourceView *D3D11DeviceExt::HandleToSrvNVX(uint32_t handle) {
+    auto mapslock = m_multithreadMapsLock.AcquireLock();
+    auto got = m_SrvHandleToPtr.find(handle);
+    if (got == m_SrvHandleToPtr.end())
+      return nullptr;
+    return (ID3D11ShaderResourceView *)got->second;
+  }
+
+
+  bool STDMETHODCALLTYPE D3D11DeviceExt::GetCudaTextureObjectNVX(uint32_t srvDriverHandle, uint32_t samplerDriverHandle, uint32_t* pCudaTextureHandle) {
+    ID3D11ShaderResourceView* srv = HandleToSrvNVX(srvDriverHandle);
+    if (!srv) {
+      Logger::warn(str::format("GetCudaTextureObjectNVX() failure - srv handle wasn't found: ", srvDriverHandle));
+      return false;
+    }
+
+    ID3D11SamplerState* samplerstate = HandleToSamplerNVX(samplerDriverHandle);
+    if (!samplerstate) {
+      Logger::warn(str::format("GetCudaTextureObjectNVX() failure - sampler handle wasn't found: ", samplerDriverHandle));
+      return false;
+    }
+
+    D3D11SamplerState* pSS = static_cast<D3D11SamplerState*>(samplerstate);
+    Rc<DxvkSampler> pDSS = pSS->GetDXVKSampler();
+    VkSampler vk_sampler = pDSS->handle();
+
+    D3D11ShaderResourceView* pSRV = static_cast<D3D11ShaderResourceView*>(srv);
+    Rc<DxvkImageView> pIV = pSRV->GetImageView();
+    VkImageView vk_imageview = pIV->handle();
+
+    VkImageViewHandleInfoNVX imageViewHandleInfo = {VK_STRUCTURE_TYPE_IMAGE_VIEW_HANDLE_INFO_NVX};
+    imageViewHandleInfo.imageView = vk_imageview;
+    imageViewHandleInfo.sampler = vk_sampler;
+    imageViewHandleInfo.descriptorType = VK_DESCRIPTOR_TYPE_COMBINED_IMAGE_SAMPLER;
+
+    // note: there's no implicit lifetime management here; it's up to the
+    // app to keep the sampler and SRV alive as long as it wants to use this
+    // derived handle.
+
+    VkDevice vkDevice = m_device->GetDXVKDevice()->handle();
+    *pCudaTextureHandle = m_device->GetDXVKDevice()->vkd()->vkGetImageViewHandleNVX(vkDevice, &imageViewHandleInfo);
+
+    if (!*pCudaTextureHandle) {
+      Logger::warn("GetCudaTextureObjectNVX() handle==0 - failed");
+      return false;
+    }
+
+    return true;
+  }
+  
+
+  bool STDMETHODCALLTYPE D3D11DeviceExt::CreateCubinComputeShaderWithNameNVX(const void* pCubin, uint32_t size, uint32_t blockX, uint32_t blockY, uint32_t blockZ, const char* pShaderName, IUnknown** phShader) {
+    dxvk::Rc<dxvk::DxvkDevice> dxvkdevice = m_device->GetDXVKDevice();
+    VkDevice vkDevice = dxvkdevice->handle();
+
+    VkCuModuleCreateInfoNVX moduleCreateInfo = { VK_STRUCTURE_TYPE_CU_MODULE_CREATE_INFO_NVX };
+    moduleCreateInfo.pData = pCubin;
+    moduleCreateInfo.dataSize = size;
+
+    VkCuModuleNVX Module;
+    VkCuFunctionNVX Function;
```

Comment:
no caps please, i know module is a keyword now but its conditional and should work here, but MSVC is ass and doesnt implement it properly. so call this cuModule and cuFunction